src/function-builder.ts

```diff
@@ -32,72 +32,112 @@ import * as https from './providers/https';
 import * as pubsub from './providers/pubsub';
 import * as remoteConfig from './providers/remoteConfig';
 import * as storage from './providers/storage';
-import {
-  CloudFunction,
-  EventContext,
-  Runnable,
-  TriggerAnnotated,
-  Schedule,
-} from './cloud-functions';
+import { CloudFunction, EventContext, Schedule } from './cloud-functions';
 
 /**
- * Configure the regions that the function is deployed to.
- * @param regions One of more region strings.
- * For example: `functions.region('us-east1')` or `functions.region('us-east1', 'us-central1')`
+ * List of all regions supported by Cloud Functions.
  */
-export function region(...regions: string[]) {
-  if (!regions.length) {
-    throw new Error('You must specify at least one region');
+const SUPPORTED_REGIONS = [
```

Comment:
Optional: this might be worth doing more as an Enum or something, but that's some additional work that can follow later...

Reply:
Noted, good idea. Will do in another PR :)